imap_processing/lo/l0/science_direct_events.py

```diff
@@ -0,0 +1,202 @@
+    TOF1: float
+    TOF2: float
+    TOF3: float
+    TIME: float
+    ENERGY: float
+    POS: float
+    case_number: int
+    tof_calculation_binary: dict
+    tof_decoder: list
+    remaining_bits: dict
+    parsed_bits: list
+
+    def __init__(self, packet, software_version: str, packet_file_name: str):
+        super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
+        self.parse_data(packet)
+
+    def decompress_data(self):
+        """Decompress the Lo Science Direct Events data."""
+        self._find_decompression_case()
+        self._find_tof_decoder_for_case()
+        self._read_tof_calculation_table()
+        self._find_remaining_bits()
+        self._parse_binary()
+        self._decode_fields()
+
+    def _find_tof_decoder_for_case(self):
+        """Get the TOF decoder for this DE's case number."""
+        # Case 0 can either be a gold or silver triple. Gold triples do
+        # not send down the TOF1 value and instead recover the TOF1 value
+        # on the ground using the decompressed checksum
+        if self.case_number == 0:
+            gold_or_silver = int(self.DATA.bin[4])
+            # The case number determines which tof decoder to use.
+            # This table shows how the TOF bits should be parsed in the
+            # binary data.
+            self.tof_decoder = ct.tof_decoder_table[self.case_number][gold_or_silver]
+
+        # Cases 4, 6, 10, 12, 13 may be Bronze triples. If it's not a bronze triple,
+        # the Position is not transmitted, but TOF3 is.
+        elif self.case_number in [4, 6, 10, 12, 13]:
+            is_bronze = int(self.DATA.bin[4])
+            self.tof_decoder = ct.tof_decoder_table[self.case_number]
+            # TODO: maybe I should just add a sub-dictionary to the tof decoder table
+            # instead of changing the values here?
+            if not is_bronze:
+                # if Bronze = 1, use table as is
+                # if Bronze = 0, position not transmitted, TOF3 is transmitted
+                self.tof_decoder["POS"] = 0
+                self.tof_decoder["TOF3"] = 6
+        # We're not expecting to recieve data for the rest of the cases, but need
+        # to handle them in case things change in the future.
+        else:
+            self.tof_decoder = ct.tof_decoder_table[self.case_number]
+
+    def _read_tof_calculation_table(self):
+        """Get the TOF calculation values for this DE's case number."""
+        # The case number determines what calculation value is needed to help
+        # calculate the TOF values
+        tof_calculation_values = ct.tof_calculation_table[self.case_number]
+        self.tof_calculation_binary = {
+            field: bitstring.Bits(calculation_value)
+            for field, calculation_value in tof_calculation_values.items()
+        }
+
+    def _find_decompression_case(self):
+        """Find the decompression case for this DE."""
+        # The first 4 bits of the binary data are used to
```

Comment:
In general the documentation for these methods is good, but the paragraphs of `#` comments should just be moved into the docstring proper. I'd also recommend saying it sets a class variable if that's the case, as that's a side affect of calling the method that should be documented.

Reply:
Good points. I'll make those updates.